CMakeLists.txt

```
@@ -0,0 +1,31 @@
+cmake_minimum_required(VERSION 3.20)
+
+# Must be done first
+if (APPLE)
+    # Force clang compiler on macOS
+    set(CMAKE_CXX_COMPILER "/usr/bin/clang++")
+    set(CMAKE_C_COMPILER "/usr/bin/clang")
+endif ()
+
+# Options
+option(WITH_TLS "Build valkey-server with TLS support" OFF)
```

Would `USE` be a more conventional word? 

```suggestion
option(USE_TLS "Build valkey-server with TLS support" OFF)
```

---

From the `configure` world, I mostly seen:

For boolean flags:

`--enable-XXX`, `--disable-XXX` (e.g. --enable-shared) etc

For flags with values, it is conventional to use `--with-XXXX` for example:

```
./configure --with-jemalloc-prefix=<prefix> ...
```

In any case, I removed this line completely as it is no longer a boolean option (it can accept: `module`, `yes` or `no`)